test/e2e/agentplatform.go

```diff
@@ -0,0 +1,210 @@
+package e2e
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+
+	commonos "github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2params"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type agentFlavor string
+
+func (af *agentFlavor) String() string {
+	return string(*af)
+}
+
+func (af *agentFlavor) Set(value string) error {
+	if len(*af) > 0 {
+		return fmt.Errorf("flavor flag already set to %s while trying to set to %s", *af, value)
+	}
+	fmt.Printf("Setting flavor flag to %s\n", value)
+	*af = agentFlavor(value)
+	return nil
+}
+
+const (
+	scriptURL    = "https://dd-agent.s3.amazonaws.com/custom"
```

Comment:
It means we might state somewhere we need to push the install_scripts to be tested in the `custom` folder prior launching the tests, correct?

Reply:
True, will change it in a follow up PR to take a param or an env var

Reply:
Actually will move it in this PR to a flag